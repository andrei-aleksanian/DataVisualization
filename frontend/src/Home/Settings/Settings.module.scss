.index {
  grid-area: settings;
  background-color: var(--color-background);
  overflow-y: scroll;
}

.Settings {
  padding: 3rem 2rem;
  color: var(--color-text);
}

.index h1 {
  margin: 0 0 1.5rem 0;
}

.submit {
  margin: 1rem 0 0 0;
}
