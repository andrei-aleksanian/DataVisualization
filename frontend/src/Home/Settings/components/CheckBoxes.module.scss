.index {
  margin: 1rem 0;
}

.index p {
  font-size: 1.1rem;
}

.CheckBoxes {
  width: 100%;
  margin: 1rem 0 var(--margin-between-elements) 0;
}

.CheckBoxes label {
  position: relative;
  cursor: pointer;
}

.CheckBoxes label:nth-child(2) {
  margin: 0.5rem 2rem;
}

.CheckBoxes label input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}

// Custom checkbox
.checkmark {
  position: absolute;
  top: 0;
  left: 0;
  height: 15px;
  width: 15px;
  background-color: var(--color-text);
  border: 1px solid var(--color-action);
  border-radius: 3px;
}

.CheckBoxes label input:checked ~ .checkmark {
  background-color: var(--color-action);
}

.text {
  margin: 0 0 0 2rem;
}
