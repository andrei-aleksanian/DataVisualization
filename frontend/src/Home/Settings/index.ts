export { default } from './Settings';
export * from './Settings';
export * from './components/ANGEL';
export * from './components/COVA';
