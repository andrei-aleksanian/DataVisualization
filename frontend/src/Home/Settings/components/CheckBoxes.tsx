import { Algorithm } from 'types/Settings';
import classes from './CheckBoxes.module.scss';

export interface CheckBoxesProps {
  currentAlgorithm: Algorithm;
  onClickAlgorithm: (event: React.ChangeEvent, algorithm: Algorithm) => void;
}

const CheckBoxes = ({ currentAlgorithm, onClickAlgorithm }: CheckBoxesProps) => (
  <div className={classes.index}>
    <p>Algorithm:</p>
    <div className={classes.CheckBoxes}>
      <label>
        <input
          type="radio"
          checked={currentAlgorithm === Algorithm.COVA}
          onChange={(e) => onClickAlgorithm(e, Algorithm.COVA)}
        />
        <span className={classes.text}>COVA</span>
        <span className={classes.checkmark} />
      </label>
      <label>
        <input
          type="radio"
          checked={currentAlgorithm === Algorithm.ANGEL}
          onChange={(e) => onClickAlgorithm(e, Algorithm.ANGEL)}
        />
        <span className={classes.text}>ANGEL</span>
        <span className={classes.checkmark} />
      </label>
    </div>
  </div>
);

export default CheckBoxes;
