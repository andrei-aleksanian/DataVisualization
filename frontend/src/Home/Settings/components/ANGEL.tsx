import Slider from 'Components/Forms/Slider';

import CheckBoxes from 'Components/Forms/CheckBoxes';

export const TEXT_SLIDER_ANCHOR_DENSITY = 'Anchor density:';
export const TEXT_SLIDER_EPSILON = 'Epsilon:';
export const TEXT_CHECKBOX_ANCHOR_MODIFICATION = 'Anchor modification:';

export enum AnchorModification {
  ON,
  OFF,
}
export interface SettingsANGEL {
  anchorDensity: number;
  epsilon: number;
  anchorModification: AnchorModification;
}
export const defaultSettingsANGEL: SettingsANGEL = {
  anchorDensity: 0.1,
  epsilon: 0.1,
  anchorModification: AnchorModification.OFF,
};
export interface SettingsANGELProps {
  settingsANGEL: SettingsANGEL;
  setSettingsANGEL: React.Dispatch<React.SetStateAction<SettingsANGEL>>;
  isCustomDataPage: boolean;
}

const ANGEL = ({ settingsANGEL, setSettingsANGEL, isCustomDataPage }: SettingsANGELProps) => {
  const onChangeAnchorDensity = (value: number) => setSettingsANGEL((prev) => ({ ...prev, anchorDensity: value }));
  const onChangeEpsilon = (value: number) => setSettingsANGEL((prev) => ({ ...prev, epsilon: value }));

  const onChangeAnchorModification = (event: React.ChangeEvent, value: AnchorModification) => {
    event.preventDefault();
    setSettingsANGEL((prev) => ({ ...prev, anchorModification: value }));
  };

  return (
    <>
      <Slider
        min={isCustomDataPage ? 0.1 : 0.05}
        max={0.2}
        step={null}
        marksArr={isCustomDataPage ? [0.1, 0.2] : [0.05, 0.1, 0.2]}
        onChange={onChangeAnchorDensity}
        text={TEXT_SLIDER_ANCHOR_DENSITY}
        value={settingsANGEL.anchorDensity}
      />
      <Slider
        min={0.1}
        max={5}
        step={null}
        marksArr={[0.1, 5]}
        onChange={onChangeEpsilon}
        text={TEXT_SLIDER_EPSILON}
        value={settingsANGEL.epsilon}
      />
      <CheckBoxes
        heading={TEXT_CHECKBOX_ANCHOR_MODIFICATION}
        currentValue={settingsANGEL.anchorModification}
        onChange={onChangeAnchorModification}
        entries={[
          { value: AnchorModification.OFF, text: '0' },
          { value: AnchorModification.ON, text: '1' },
        ]}
      />
    </>
  );
};

export default ANGEL;
