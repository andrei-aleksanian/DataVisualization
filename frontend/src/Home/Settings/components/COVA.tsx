import Slider from 'Components/Forms/Slider';

import CheckBoxes from 'Components/Forms/CheckBoxes';

export const H1_TEXT = 'Example: Cylinder';

export enum C {
  ORIGINAL,
  PERCENTAGE,
}

interface SettingsCOVA {
  alpha: number;
  c: C;
}

export const defaultSettingsCOVA: SettingsCOVA = {
  alpha: 0,
  c: C.ORIGINAL,
};

export interface SettingsCOVAProps {
  settingsCOVA: SettingsCOVA;
  setSettingsCOVA: React.Dispatch<React.SetStateAction<SettingsCOVA>>;
}

const COVA = ({ settingsCOVA, setSettingsCOVA }: SettingsCOVAProps) => {
  const onChangeAlpha = (value: number) => setSettingsCOVA((prev) => ({ ...prev, alpha: value }));
  const onChangeC = (event: React.ChangeEvent, value: C) => {
    event.preventDefault();
    setSettingsCOVA((prev) => ({ ...prev, c: value }));
  };

  return (
    <>
      <Slider
        min={0}
        max={1}
        step={0.2}
        marksArr={[0, 0.2, 0.4, 0.6, 0.8, 1]}
        onChange={onChangeAlpha}
        text="Alpha"
        value={settingsCOVA.alpha}
      />
      <CheckBoxes
        heading="C parameter:"
        currentValue={settingsCOVA.c}
        onChange={onChangeC}
        entries={[
          { value: C.ORIGINAL, text: 'Original label' },
          { value: C.PERCENTAGE, text: '10% of the number of points' },
        ]}
      />
    </>
  );
};

export default COVA;
