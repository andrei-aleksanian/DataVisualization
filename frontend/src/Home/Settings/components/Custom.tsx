import FileDropArea, { FileArea } from 'Components/Forms/FileDropArea';
import Button from 'Components/Forms/Button';
import CheckBoxes from 'Components/Forms/CheckBoxes';
import Error from 'Components/Forms/Error';

import classes from '../Settings.module.scss';

export enum Dimension {
  D2,
  D3,
}
export interface SettingsCustom {
  dimension: Dimension;
  file: FileArea;
}
export const defaultSettingsCustom: SettingsCustom = {
  file: { file: null, error: null },
  dimension: Dimension.D2,
};
export interface CustomProps {
  onSubmit: Function;
  setSettingsCustom: React.Dispatch<React.SetStateAction<SettingsCustom>>;
  settingsCustom: SettingsCustom;
  error: string | null;
  acceptedType: string;
}

export const TEXT_CHECKBOX_DIMENSION = 'Dimension:';
export const TEXT_BUTTON = 'Submit';

const Custom = ({
  onSubmit,
  setSettingsCustom,
  settingsCustom: { dimension, file },
  error,
  acceptedType,
}: CustomProps) => {
  const onChangeDimension = (event: React.ChangeEvent, value: Dimension) => {
    event.preventDefault();
    setSettingsCustom((prev) => ({ ...prev, dimension: value }));
  };
  const setFile = (fileNew: FileArea) => {
    setSettingsCustom((prev) => ({ ...prev, file: fileNew }));
  };

  return (
    <div>
      <CheckBoxes
        heading={TEXT_CHECKBOX_DIMENSION}
        currentValue={dimension}
        onChange={onChangeDimension}
        entries={[
          { value: Dimension.D2, text: '2' },
          { value: Dimension.D3, text: '3' },
        ]}
      />
      <FileDropArea setFile={setFile} file={file} acceptedType={acceptedType} />
      <Button text={TEXT_BUTTON} onClick={() => onSubmit()} active={false} customClass={classes.submit} />
      {error && <Error text={error} />}
    </div>
  );
};

export default Custom;
