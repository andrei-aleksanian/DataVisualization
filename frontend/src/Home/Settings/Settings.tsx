import Slider from 'Components/Forms/Slider';
import { useState } from 'react';
import { LinkBack } from 'Components/Link';

import { Algorithm } from 'types/Settings';
import CheckBoxes from 'Components/Forms/CheckBoxes';

import classes from './Settings.module.scss';
import COVA, { defaultSettingsCOVA } from './components/COVA';
import ANGEL, { defaultSettingsANGEL } from './components/ANGEL';

export const TEXT_CHECKBOX_COVA = 'COVA';
export const TEXT_CHECKBOX_ANGEL = 'ANGEL';
export const TEXT_H1 = 'Example: Cylinder';
export const TEXT_CHECKBOX_ALGORITHM = 'Algorithm:';
export const TEXT_SLIDER_NEIGHBOUR = 'Neighbour:';
export const TEXT_SLIDER_LAMBDA = 'Lambda:';
export const TEXT_CHECKBOX_PRESERVATION = 'Show Data Preservation:';

export enum DataPreservation {
  ON,
  OFF,
}
export interface SettingsCommon {
  dataPreservation: DataPreservation;
  algorithm: Algorithm;
  neighbour: number;
  lambda: number;
}
export const defaultSettingsCommon = {
  dataPreservation: DataPreservation.OFF,
  algorithm: Algorithm.COVA,
  neighbour: 0,
  lambda: 0,
};

const Settings = () => {
  const [settingsCommon, setSettingsCommon] = useState(defaultSettingsCommon);
  const [settingsCOVA, setSettingsCOVA] = useState(defaultSettingsCOVA);
  const [settingsANGEL, setSettingsANGEL] = useState(defaultSettingsANGEL);

  const onChangeAlgorithm = (event: React.ChangeEvent, newAlgorithm: Algorithm) => {
    event.preventDefault();
    setSettingsCommon((prev) => ({ ...prev, algorithm: newAlgorithm }));
  };
  const onChangeNeighbour = (value: number) => setSettingsCommon((prev) => ({ ...prev, neighbour: value }));
  const onChangeLambda = (value: number) => setSettingsCommon((prev) => ({ ...prev, lambda: value }));
  const onChangeDataPreservation = (event: React.ChangeEvent, value: DataPreservation) => {
    event.preventDefault();
    setSettingsCommon((prev) => ({ ...prev, dataPreservation: value }));
  };

  return (
    <div className={classes.index}>
      <LinkBack link="/" />
      <h1>{TEXT_H1}</h1>
      <CheckBoxes
        heading={TEXT_CHECKBOX_ALGORITHM}
        currentValue={settingsCommon.algorithm}
        onChange={onChangeAlgorithm}
        entries={[
          { value: Algorithm.COVA, text: TEXT_CHECKBOX_COVA },
          { value: Algorithm.ANGEL, text: TEXT_CHECKBOX_ANGEL },
        ]}
      />
      <Slider
        min={0}
        max={7}
        step={1}
        marksArr={['10', '20', '30', '10%', '20%', '30%', '40%', '50%']}
        onChange={onChangeNeighbour}
        text={TEXT_SLIDER_NEIGHBOUR}
        value={settingsCommon.neighbour}
      />
      <Slider
        min={0}
        max={1}
        step={0.2}
        marksArr={[0, 0.2, 0.4, 0.6, 0.8, 1]}
        onChange={onChangeLambda}
        text={TEXT_SLIDER_LAMBDA}
        value={settingsCommon.lambda}
      />
      {settingsCommon.algorithm === Algorithm.COVA ? (
        <COVA {...{ settingsCOVA, setSettingsCOVA }} />
      ) : (
        <ANGEL {...{ settingsANGEL, setSettingsANGEL }} />
      )}
      <CheckBoxes
        heading={TEXT_CHECKBOX_PRESERVATION}
        currentValue={settingsCommon.dataPreservation}
        onChange={onChangeDataPreservation}
        entries={[
          { value: DataPreservation.ON, text: 'Yes' },
          { value: DataPreservation.OFF, text: 'No' },
        ]}
      />
    </div>
  );
};

export default Settings;
