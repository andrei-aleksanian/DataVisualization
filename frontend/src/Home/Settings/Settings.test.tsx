import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { Algorithm } from 'types/Settings';
import { DATA_TESTID_MODAL } from 'Components/UI';
import Settings, {
  TEXT_CHECKBOX_ALGORITHM,
  TEXT_CHECKBOX_PRESERVATION,
  TEXT_SLIDER_NEIGHBOUR,
  TEXT_CHECKBOX_ANGEL,
  TEXT_CHECKBOX_COVA,
  TEXT_CHECKBOX_ORIGINAL,
  defaultSettingsCommon,
  SettingsProps,
} from '.';
import { TEXT_CHECKBOX_COHORT_NUMBER, TEXT_SLIDER_ALPHA, defaultSettingsCOVA } from './components/COVA';
import {
  TEXT_SLIDER_ANCHOR_DENSITY,
  TEXT_CHECKBOX_ANCHOR_MODIFICATION,
  TEXT_SLIDER_EPSILON,
  defaultSettingsANGEL,
} from './components/ANGEL';
import { defaultSettingsCustom } from './components/Custom';

const mockSettingsProps: SettingsProps = {
  settingsCommon: defaultSettingsCommon,
  setSettingsCommon: () => {},
  settingsCOVA: defaultSettingsCOVA,
  setSettingsCOVA: () => {},
  settingsANGEL: defaultSettingsANGEL,
  setSettingsANGEL: () => {},
  backLink: '/examples',
  reviewer: false,
  name: 'test',
};

const getSettingsProps = (reviewer: boolean = false): SettingsProps => {
  const props: SettingsProps = reviewer
    ? {
        ...mockSettingsProps,
        customDataPage: {
          setSettingsCustom: () => {},
          settingsCustom: defaultSettingsCustom,
          acceptedType: '.mat',
        },
      }
    : {
        ...mockSettingsProps,
        reviewer: true,
      };
  return props;
};

test('Settings matches snapshot', () => {
  const { asFragment } = render(<Settings {...mockSettingsProps} />, { wrapper: MemoryRouter });
  expect(asFragment()).toMatchSnapshot();
});

describe('Test UI changes and events', () => {
  const init = (algorithm: Algorithm) => {
    const mockSettingsPropsInit: SettingsProps = {
      ...mockSettingsProps,
      settingsCommon: {
        ...mockSettingsProps.settingsCommon,
        algorithm, // pretending we clicked the button in parent
      },
    };
    return render(<Settings {...mockSettingsPropsInit} />, { wrapper: MemoryRouter });
  };

  const checkCommonArttributes = (asFragment: () => DocumentFragment) => {
    expect(screen.getByText(TEXT_CHECKBOX_ALGORITHM)).toBeInTheDocument();
    expect(screen.getByText(TEXT_SLIDER_NEIGHBOUR)).toBeInTheDocument();
    expect(screen.getByText(TEXT_CHECKBOX_PRESERVATION)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  };

  test("Checkboxes clicks don't change common attributes", () => {
    const { asFragment } = init(Algorithm.ANGEL);
    checkCommonArttributes(asFragment);
  });

  test("Checkboxes clicks don't change common attributes", () => {
    const { asFragment } = init(Algorithm.COVA);
    checkCommonArttributes(asFragment);
  });

  test('Checkboxes ANGEL/COVA change the controls to ANGEL', () => {
    init(Algorithm.ANGEL);

    expect(screen.getByRole('radio', { name: TEXT_CHECKBOX_COVA })).not.toBeChecked();
    expect(screen.getByRole('radio', { name: TEXT_CHECKBOX_ANGEL })).toBeChecked();

    // ANGEL Elements should be rendered
    expect(screen.getByText(TEXT_SLIDER_ANCHOR_DENSITY)).toBeInTheDocument();
    expect(screen.getByText(TEXT_SLIDER_EPSILON)).toBeInTheDocument();
    expect(screen.getByText(TEXT_CHECKBOX_ANCHOR_MODIFICATION)).toBeInTheDocument();

    // COVA elements shouldn't be rendered
    expect(screen.queryByText(TEXT_SLIDER_ALPHA)).not.toBeInTheDocument();
    expect(screen.queryByText(TEXT_CHECKBOX_COHORT_NUMBER)).not.toBeInTheDocument();
  });

  test('Checkboxes ANGEL/COVA change the controls back to COVA', async () => {
    init(Algorithm.COVA);

    expect(screen.getByRole('radio', { name: TEXT_CHECKBOX_ANGEL })).not.toBeChecked();
    expect(screen.getByRole('radio', { name: TEXT_CHECKBOX_COVA })).toBeChecked();

    // COVA Elements should be rendered
    expect(screen.getByText(TEXT_SLIDER_ALPHA)).toBeInTheDocument();
    expect(screen.getByText(TEXT_CHECKBOX_COHORT_NUMBER)).toBeInTheDocument();

    // ANGEL elements shouldn't be rendered
    expect(screen.queryByText(TEXT_SLIDER_ANCHOR_DENSITY)).not.toBeInTheDocument();
    expect(screen.queryByText(TEXT_SLIDER_EPSILON)).not.toBeInTheDocument();
    expect(screen.queryByText(TEXT_CHECKBOX_ANCHOR_MODIFICATION)).not.toBeInTheDocument();
  });

  test('Checkboxes ANGEL/COVA change the controls to ORIGINAL', async () => {
    init(Algorithm.ORIGINAL);

    expect(screen.getByRole('radio', { name: TEXT_CHECKBOX_ANGEL })).not.toBeChecked();
    expect(screen.getByRole('radio', { name: TEXT_CHECKBOX_COVA })).not.toBeChecked();
    expect(screen.getByRole('radio', { name: TEXT_CHECKBOX_ORIGINAL })).toBeChecked();

    // Modal should be rendered above all controls
    expect(screen.getByTestId(DATA_TESTID_MODAL)).toBeInTheDocument();
  });
});

test('Settings for Custom page matches snapshot', () => {
  const { asFragment } = render(<Settings {...getSettingsProps()} />, { wrapper: MemoryRouter });
  expect(asFragment()).toMatchSnapshot();
});

test('Settings for Reviewer page matches snapshot', () => {
  const { asFragment } = render(<Settings {...getSettingsProps(true)} />, { wrapper: MemoryRouter });
  expect(asFragment()).toMatchSnapshot();
});
