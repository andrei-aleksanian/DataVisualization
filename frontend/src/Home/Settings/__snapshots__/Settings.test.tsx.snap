// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="index"
  >
    <a
      class="index Back"
      href="/examples"
    >
      <img
        alt="back"
        src="arrow.svg"
      />
    </a>
    <h1>
      Example: Cylinder
    </h1>
    <div
      class="index"
    >
      <p>
        Algorithm:
      </p>
      <div
        class="CheckBoxes"
      >
        <label>
          <input
            checked=""
            type="radio"
          />
          <span
            class="text"
          >
            COVA
          </span>
          <span
            class="checkmark"
          />
        </label>
        <label>
          <input
            type="radio"
          />
          <span
            class="text"
          >
            ANGEL
          </span>
          <span
            class="checkmark"
          />
        </label>
      </div>
    </div>
    <div
      class="index"
    >
      <p>
        Neighbour number:
      </p>
      <div
        class="rc-slider rc-slider-with-marks"
      >
        <div
          class="rc-slider-rail"
          style="height: 4px;"
        />
        <div
          class="rc-slider-track"
          style="height: 4px; left: 0%; width: 0%;"
        />
        <div
          class="rc-slider-step"
        >
          <span
            class="rc-slider-dot rc-slider-dot-active"
            style="height: 10px; width: 10px; bottom: -3px; left: 0%;"
          />
          <span
            class="rc-slider-dot"
            style="border: 2px solid #fff; height: 10px; width: 10px; bottom: -3px; left: 20%;"
          />
          <span
            class="rc-slider-dot"
            style="border: 2px solid #fff; height: 10px; width: 10px; bottom: -3px; left: 40%;"
          />
          <span
            class="rc-slider-dot"
            style="border: 2px solid #fff; height: 10px; width: 10px; bottom: -3px; left: 60%;"
          />
          <span
            class="rc-slider-dot"
            style="border: 2px solid #fff; height: 10px; width: 10px; bottom: -3px; left: 80%;"
          />
          <span
            class="rc-slider-dot"
            style="border: 2px solid #fff; height: 10px; width: 10px; bottom: -3px; left: 100%;"
          />
        </div>
        <div
          aria-disabled="false"
          aria-valuemax="5"
          aria-valuemin="0"
          aria-valuenow="0"
          class="rc-slider-handle"
          role="slider"
          style="height: 20px; width: 20px; margin-top: -9px; box-shadow: none; left: 0%; transform: translateX(-50%);"
          tabindex="0"
        />
        <div
          class="rc-slider-mark"
        >
          <span
            class="rc-slider-mark-text rc-slider-mark-text-active"
            style="transform: translateX(-50%); left: 0%; font-size: 0.85rem; color: rgb(204, 204, 204);"
          >
            10
          </span>
          <span
            class="rc-slider-mark-text"
            style="transform: translateX(-50%); left: 20%; font-size: 0.85rem; color: rgb(204, 204, 204);"
          >
            20
          </span>
          <span
            class="rc-slider-mark-text"
            style="transform: translateX(-50%); left: 40%; font-size: 0.85rem; color: rgb(204, 204, 204);"
          >
            30
          </span>
          <span
            class="rc-slider-mark-text"
            style="transform: translateX(-50%); left: 60%; font-size: 0.85rem; color: rgb(204, 204, 204);"
          >
            10%
          </span>
          <span
            class="rc-slider-mark-text"
            style="transform: translateX(-50%); left: 80%; font-size: 0.85rem; color: rgb(204, 204, 204);"
          >
            30%
          </span>
          <span
            class="rc-slider-mark-text"
            style="transform: translateX(-50%); left: 100%; font-size: 0.85rem; color: rgb(204, 204, 204);"
          >
            50%
          </span>
        </div>
      </div>
    </div>
    <div
      class="index"
    >
      <p>
        Lambda:
      </p>
      <div
        class="rc-slider rc-slider-with-marks"
      >
        <div
          class="rc-slider-rail"
          style="height: 4px;"
        />
        <div
          class="rc-slider-track"
          style="height: 4px; left: 0%; width: 0%;"
        />
        <div
          class="rc-slider-step"
        >
          <span
            class="rc-slider-dot rc-slider-dot-active"
            style="height: 10px; width: 10px; bottom: -3px; left: 0%;"
          />
          <span
            class="rc-slider-dot"
            style="border: 2px solid #fff; height: 10px; width: 10px; bottom: -3px; left: 20%;"
          />
          <span
            class="rc-slider-dot"
            style="border: 2px solid #fff; height: 10px; width: 10px; bottom: -3px; left: 40%;"
          />
          <span
            class="rc-slider-dot"
            style="border: 2px solid #fff; height: 10px; width: 10px; bottom: -3px; left: 60%;"
          />
          <span
            class="rc-slider-dot"
            style="border: 2px solid #fff; height: 10px; width: 10px; bottom: -3px; left: 80%;"
          />
          <span
            class="rc-slider-dot"
            style="border: 2px solid #fff; height: 10px; width: 10px; bottom: -3px; left: 100%;"
          />
        </div>
        <div
          aria-disabled="false"
          aria-valuemax="1"
          aria-valuemin="0"
          aria-valuenow="0"
          class="rc-slider-handle"
          role="slider"
          style="height: 20px; width: 20px; margin-top: -9px; box-shadow: none; left: 0%; transform: translateX(-50%);"
          tabindex="0"
        />
        <div
          class="rc-slider-mark"
        >
          <span
            class="rc-slider-mark-text rc-slider-mark-text-active"
            style="transform: translateX(-50%); left: 0%; font-size: 0.85rem; color: rgb(204, 204, 204);"
          >
            0
          </span>
          <span
            class="rc-slider-mark-text"
            style="transform: translateX(-50%); left: 20%; font-size: 0.85rem; color: rgb(204, 204, 204);"
          >
            0.2
          </span>
          <span
            class="rc-slider-mark-text"
            style="transform: translateX(-50%); left: 40%; font-size: 0.85rem; color: rgb(204, 204, 204);"
          >
            0.4
          </span>
          <span
            class="rc-slider-mark-text"
            style="transform: translateX(-50%); left: 60%; font-size: 0.85rem; color: rgb(204, 204, 204);"
          >
            0.6
          </span>
          <span
            class="rc-slider-mark-text"
            style="transform: translateX(-50%); left: 80%; font-size: 0.85rem; color: rgb(204, 204, 204);"
          >
            0.8
          </span>
          <span
            class="rc-slider-mark-text"
            style="transform: translateX(-50%); left: 100%; font-size: 0.85rem; color: rgb(204, 204, 204);"
          >
            1
          </span>
        </div>
      </div>
    </div>
    <div
      class="index"
    >
      <p>
        Alpha:
      </p>
      <div
        class="rc-slider rc-slider-with-marks"
      >
        <div
          class="rc-slider-rail"
          style="height: 4px;"
        />
        <div
          class="rc-slider-track"
          style="height: 4px; left: 0%; width: 40%;"
        />
        <div
          class="rc-slider-step"
        >
          <span
            class="rc-slider-dot rc-slider-dot-active"
            style="height: 10px; width: 10px; bottom: -3px; left: 0%;"
          />
          <span
            class="rc-slider-dot rc-slider-dot-active"
            style="height: 10px; width: 10px; bottom: -3px; left: 20%;"
          />
          <span
            class="rc-slider-dot rc-slider-dot-active"
            style="height: 10px; width: 10px; bottom: -3px; left: 40%;"
          />
          <span
            class="rc-slider-dot"
            style="border: 2px solid #fff; height: 10px; width: 10px; bottom: -3px; left: 60%;"
          />
          <span
            class="rc-slider-dot"
            style="border: 2px solid #fff; height: 10px; width: 10px; bottom: -3px; left: 80%;"
          />
          <span
            class="rc-slider-dot"
            style="border: 2px solid #fff; height: 10px; width: 10px; bottom: -3px; left: 100%;"
          />
        </div>
        <div
          aria-disabled="false"
          aria-valuemax="1"
          aria-valuemin="0"
          aria-valuenow="0.4"
          class="rc-slider-handle"
          role="slider"
          style="height: 20px; width: 20px; margin-top: -9px; box-shadow: none; left: 40%; transform: translateX(-50%);"
          tabindex="0"
        />
        <div
          class="rc-slider-mark"
        >
          <span
            class="rc-slider-mark-text rc-slider-mark-text-active"
            style="transform: translateX(-50%); left: 0%; font-size: 0.85rem; color: rgb(204, 204, 204);"
          >
            0
          </span>
          <span
            class="rc-slider-mark-text rc-slider-mark-text-active"
            style="transform: translateX(-50%); left: 20%; font-size: 0.85rem; color: rgb(204, 204, 204);"
          >
            0.2
          </span>
          <span
            class="rc-slider-mark-text rc-slider-mark-text-active"
            style="transform: translateX(-50%); left: 40%; font-size: 0.85rem; color: rgb(204, 204, 204);"
          >
            0.4
          </span>
          <span
            class="rc-slider-mark-text"
            style="transform: translateX(-50%); left: 60%; font-size: 0.85rem; color: rgb(204, 204, 204);"
          >
            0.6
          </span>
          <span
            class="rc-slider-mark-text"
            style="transform: translateX(-50%); left: 80%; font-size: 0.85rem; color: rgb(204, 204, 204);"
          >
            0.8
          </span>
          <span
            class="rc-slider-mark-text"
            style="transform: translateX(-50%); left: 100%; font-size: 0.85rem; color: rgb(204, 204, 204);"
          >
            1
          </span>
        </div>
      </div>
    </div>
    <div
      class="index"
    >
      <p>
        Cohort number:
      </p>
      <div
        class="CheckBoxes"
      >
        <label>
          <input
            type="radio"
          />
          <span
            class="text"
          >
            Original cohort number
          </span>
          <span
            class="checkmark"
          />
        </label>
        <label>
          <input
            checked=""
            type="radio"
          />
          <span
            class="text"
          >
            10% of the number of points
          </span>
          <span
            class="checkmark"
          />
        </label>
      </div>
    </div>
    <div
      class="index"
    >
      <p>
        Show Data Preservation Error:
      </p>
      <div
        class="CheckBoxes"
      >
        <label>
          <input
            type="radio"
          />
          <span
            class="text"
          >
            Yes
          </span>
          <span
            class="checkmark"
          />
        </label>
        <label>
          <input
            checked=""
            type="radio"
          />
          <span
            class="text"
          >
            No
          </span>
          <span
            class="checkmark"
          />
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
